From oliver@neukum.name Sat Oct 28 02:55:12 2006
From: Oliver Neukum <oliver@neukum.name>
To: greg@kroah.com, toralf.foerster@gmx.de, <vojtech@suse.cz>, linux-usb-devel@lists.sourceforge.net
Subject: USB: failure in usblp's error path
Date: Sat, 28 Oct 2006 11:36:59 +0200
Content-Disposition: inline
Message-Id: <200610281136.59634.oliver@neukum.name>

if urb submission fails due to a transient error here eg. ENOMEM
, the driver is dead. This fixes it.

	Regards
		Oliver

Signed-off-by: Oliver Neukum <oliver@neukum.name>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/class/usblp.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/usb/class/usblp.c
+++ gregkh-2.6/drivers/usb/class/usblp.c
@@ -722,6 +722,7 @@ static ssize_t usblp_write(struct file *
 		usblp->wcomplete = 0;
 		err = usb_submit_urb(usblp->writeurb, GFP_KERNEL);
 		if (err) {
+			usblp->wcomplete = 1;
 			if (err != -ENOMEM)
 				count = -EIO;
 			else
